ui/src/app/customizations/api-connectors/api-connector-list.component.html

```diff
@@ -0,0 +1,6 @@
+<div class="row api-connector-list">
+  <div class="col-xs-12">
+    <h1>API Client Connector</h1>
+    <p>Nullam quis risus eget urna mollis ornare vel eu leo. Find out more about this at <a href="https://access.redhat.com/documentation/en-us/red_hat_jboss_fuse/7.0-tp/html-single/fuse_ignite_sample_integration_tutorial/" rel="nofollow" target="_blank">{{ config.getSettings('branding', 'appName', 'Syndesis') }} Help</a>.</p>
```

Comment:
@kahboom A simple but highly effective way to optimize performance in Angular applications is by not computing values in the template itself. Please take a look to the code above. We have place in the template the following function call wrapped by an interpolation binding:

```{{ config.getSettings('branding', 'appName', 'Syndesis') }}```

This entails that Angular is re-running the `getSettings(...)` method on each change detection cycle. This might become an expensive operation if the function wraps a significant logic, not to mention that this kind of bindings are less readable.

What are the best workarounds for this? There're several approaches:

* using a getter
* binding that value just once to a class field
* using a _pure_ pipe. 

My favorite take in this case would be to use a pure pie. The pure pipe will only be re-run if the input reference type changes, which is unlikely the case, and you can also re-use it in some other instances of the application. This also allows to remove logic from the underlying component controller, since there is no need to inject the `config` object anymore:

```html
<a href="blah blah" rel="nofollow" target="_blank">
  {{ 'branding' | syndesysConfig: ['appName', 'Syndesis'] }} Help
</a>
```

A `SyndesisConfigPipe` would expect a string with the setting key we want to bring (`branding`) and the pipe should be fed with the group key that setting belongs to (`appName`). Additionally, the pipe payload can accept a second argument, which is the default value.

This approach features a minimum performance cost and makes that data binding available for the entire application, leading to less code in our components.

Reply:
@deeleman is there an issue open for this?  We could totally adopt this approach, let's create an issue to expose the config via a pipe in this manner if it's not already open.

Reply:
Sorry, I completely copy-pasta'd this, but I agree that this would be a good approach. I will adjust accordingly once we can work refactoring this throughout the app. @gashcrumb I checked very briefly but couldn't find anything so I opened one here: #392